packages/dashboard/src/util/viewport.ts

```diff
@@ -0,0 +1,34 @@
+import { Viewport } from '@iot-app-kit/core';
+import { SerializableViewport } from '~/store/state';
+
+export const serializeViewport = (
+  viewport?: Viewport
+): SerializableViewport | undefined => {
+  if (!viewport) return undefined;
+
+  if ('duration' in viewport) {
+    return viewport;
+  } else {
+    return {
+      start: new Date(viewport.start).getTime(),
+      end: new Date(viewport.end).getTime(),
+      group: viewport.group,
+    };
+  }
+};
+
+export const deserializeViewport = (
+  viewport?: SerializableViewport
+): Viewport | undefined => {
+  if (!viewport) return undefined;
+
+  if ('duration' in viewport) {
+    return viewport;
+  } else {
+    return {
+      start: new Date(viewport.start),
+      end: new Date(viewport.end),
+      group: viewport.group,
+    };
+  }
+};
```

Comment:
Neither of these functions appear to be doing what their name implies. Shouldn't there be some JSON.stringify/JSON.parse involved?

Reply:
oh yeah that woudl've been easier lol. updated code :)